static/app/components/profiling/flamegraph.tsx

```diff
@@ -243,6 +246,26 @@ function Flamegraph(props: FlamegraphProps): ReactElement {
     flamegraphView,
   ]);
 
+  const flamegraphRenderer = useMemo(() => {
+    if (!flamegraphCanvasRef) {
+      return null;
+    }
+
+    return new FlamegraphRenderer(flamegraphCanvasRef, flamegraph, flamegraphTheme, {
+      draw_border: true,
+    });
+  }, [flamegraph, flamegraphCanvasRef, flamegraphTheme]);
```

Comment:
It feels kind of odd that we're lifting the renderer from the zoom view up to the flamegraph, especially since the purpose is the be able to use `formatColorForFrame`. Should we consider decouple that from the renderer since it's not strictly necessary to couple them?

I see you've already created a `useFlamegraphColors` so this might be good to do instead of lifting the whole renderer.

Reply:
I think that passing the renderer around is a mistake and we should avoid it.

```
Should we consider decouple that from the renderer since it's not strictly necessary to couple them?
```

Yes, but that may turn out a bit trickier down the line when we start adding differential charts as colors are going to be unique to each chart and we'll probably need some centralized way to query them. Once I reintroduce a differential view I will cleanup this logic and create the colors for each chart independently and pass them down and it should give me a bit of a better idea of the best way to handle this

Reply:
As discussed on slack, instead of lifting the flamegraph renderer, we should be lifting just the color map by decoupling them. But for simplicity, let's do this for now and do that in a follow up PR 👍

Reply:
Posting this here just in case anyone wants to read it.

```
Usually, the workflow with differential view is that you have a base profile and you then select what to compare it to. If we dont have it unique per chart, it would mean that when a user selects a profile to compare to, we need to repaint everything and generate new colors (both for the base and the new profile) which can be visually confusing as now all of a sudden everything is colored differently
```

So tldr, the end goal is to lower the renderer back to the component as @Zylphrex noted, but we'll do that once we know how we want the differential experience to behave - I did not want to tackle it here